### DISCUSSION TITLE
using Readme.md or index.md for main docs

### DISCUSSION THREAD

**Discussion by qtzar** [avisi-cloud/structurizr-site-generatr#204] [General]
As we roll out this tool to our engineers a few questions have come up around using the !docs directive.

Currently if !docs is added and pointed to a folder it uses the files in the folder in alphanumeric order, the first file being used to populate the main 'Info' page for a system and the rest being added to the 'documents' tab.

Our engineers have expressed a desire to use a fixed named file such as readme.md or index.md for the info page if it exists in the docs folder. Their reason for doing this is because they currently use a readme.md file in their code repositories and they are looking at using GitHub actions to copy that file to the central diagrams repository in to the correct place. Mainly they want to keep their documentation with their code and not have to manually copy it if they want to update it.

If this is a good idea I can write the code for it. I'm thinking of index.md, then readme.md and then a final fallback to the current first file in the directory, making sure to exclude the file that was used for the documentation tab.

**Comment by dirkgroot** [comment]
Thanks for bringing this up! I think that having a property for this would be the best way to be able to suit everybody's needs. We could add a property called `site-generatr.docs.index-file` (or another, better, name) at workspace-level. When this property has a value, it's being used. If not, then we fall back to the current behavior.

Would this be a good option for you?
